src/app/shell/admin-tools/data/users/users.component.ts

```diff
@@ -85,6 +88,9 @@ export class UsersComponent implements OnInit, OnDestroy {
    * @param event: MatTabChangeEvent
    */
   onTabChange(event: MatTabChangeEvent): void {
+    this.currentPage = PaginationConstants.firstPage;
+    Util.setPaginationParams(this.childrenParams, this.currentPage, this.childrenParams.size);
```

Comment:
This exact piece of code repeats all over the code, so please create a separate method and does this action

private setPagination(): void {
    this.currentPage = PaginationConstants.firstPage;
    Util.setPaginationParams(this.childrenParams, this.currentPage, this.childrenParams.size);
}

and call this method where it is needed

Reply:
because at least at this file it is repeated 3 times